internal/controller/numaflowcontrollerrollout_controller.go

```diff
@@ -84,19 +85,24 @@ func NewNumaflowControllerRolloutReconciler(
 	kubectl kubeUtil.Kubectl,
 	customMetrics *metrics.CustomMetrics,
 ) (*NumaflowControllerRolloutReconciler, error) {
-	stateCache := sync.NewLiveStateCache(rawConfig)
+	stateCache := sync.NewLiveStateCache(rawConfig, customMetrics)
+	newRawConfig := metrics.AddMetricsTransportWrapper(customMetrics, rawConfig)
```

Comment:
why do we call this function here as opposed to from main.go?

Reply:
same question for the `kubectl.SetOnKubectlRun` call below

Reply:
now I see in the [issue](https://github.com/numaproj/numaplane/issues/97) it says we're counting these only for NumaflowControllerRollout, but i'm not sure why

Reply:
Numaflow Controller is through gitops sync

Reply:
Not that we need to hold up this PR for this, but why do we only want to count "Number of kubernetes requests executed during reconciliation" for Numaflow Controller and not for ISBService and Pipeline?

Reply:
yeah I think we would want all, @chandankumar4 can you create an follow up issue to track this?

Reply:
I have covered it in this PR